### ISSUE TITLE
[RLlib] change to WorkerSet.trainable_policies in 1.12 breaks custom algorithms

### ISSUE THREAD

**Issue by cassidylaidlaw** [ray-project/ray#24082]
### What happened + What you expected to happen

In versions of RLlib up to 1.11, calling `trainer.workers.trainable_policies()` returns a list of policy IDs:

    >>> trainer.workers.trainable_policies()
    ['default_policy']

Now in 1.12, it returns a list of booleans:

    >>> trainer.workers.trainable_policies()
    [True]

I understand that this method is deprecated as of 1.12, but this breaks all the execution plans of my custom algorithms due to how `SelectExperiences` works. Most of my custom algorithms (as well as RLlib-provided ones) have a line in their execution plans like

    rollouts = rollouts.for_each(SelectExperiences(workers.trainable_policies()))

I've noticed that in version 1.12, this has been updated in the algorithms that ship with RLlib to

    rollouts = rollouts.for_each(SelectExperiences(local_worker=workers.local_worker()))

However, there's no easy way to support both versions! Even though the 1.12 `SelectExperiences` will take both `local_worker` and `policy_ids` arguments and defer to `local_worker`, you can't pass `local_worker` to the 1.11 version of `SelectExperiences`.

It seems like in general this change (altering the return type of `trainable_policies`) was a bit weird and unnecessary. Do you have some idea of a good workaround to write code that supports both 1.11 and 1.12?

### Versions / Dependencies

Ray: 1.11.0 and 1.12.0
Python: 3.8.12
OS: Ubuntu 20.04.4

### Reproduction script

Works in RLlib 1.11.0 but not 1.12.0:

    from ray.rllib.agents.ppo import PPOTrainer
    from ray.rllib.execution.rollout_ops import ParallelRollouts, ConcatBatches, StandardizeFields, SelectExperiences
    from ray.rllib.execution.train_ops import TrainOneStep
    from ray.rllib.execution.metric_ops import StandardMetricsReporting
    from ray.rllib.examples.env.multi_agent import MultiAgentCartPole
    import random

    class MyCustomTrainer(PPOTrainer):
        @staticmethod
        def execution_plan(workers, config):
            rollouts = ParallelRollouts(workers, mode="bulk_sync")
            print(workers.trainable_policies())
            rollouts = rollouts.for_each(
                SelectExperiences(policy_ids=workers.trainable_policies())
            )
            rollouts = rollouts.combine(
                ConcatBatches(
                    min_batch_size=config["train_batch_size"],
                    count_steps_by=config["multiagent"]["count_steps_by"],
                )
            )
            rollouts = rollouts.for_each(StandardizeFields(["advantages"]))

            train_op = rollouts.for_each(
                TrainOneStep(
                    workers,
                    num_sgd_iter=config["num_sgd_iter"],
                    sgd_minibatch_size=config["sgd_minibatch_size"],
                )
            )

            # Warn about bad reward scales and return training metrics.
            return StandardMetricsReporting(train_op, workers, config)

    policy_config = (None, MultiAgentCartPole.observation_space, MultiAgentCartPole.action_space, {})
    trainer = MyCustomTrainer({
        "env": MultiAgentCartPole,
        "env_config": {"num_agents": 2},
        "framework": "torch",
        "multiagent": {
            "policies": {"policy_0": policy_config, "policy_1": policy_config},
            "policy_mapping_fn": lambda agent_id, **kwargs: random.choice(["policy_0", "policy_1"])
        },
    })
    result = trainer.train()
    assert result["info"]["num_agent_steps_trained"] > 0

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by cassidylaidlaw** [comment]
Thank you for the fix! I do understand it's deprecated, but hopefully this will make it easier to transition across versions to the new method :)
